sql/2021/performance/web_vitals_by_device.sql

```diff
@@ -0,0 +1,155 @@
+#standardSQL
+# Core WebVitals by device
+
+CREATE TEMP FUNCTION IS_GOOD (good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
+  good / (good + needs_improvement + poor) >= 0.75
+);
+
+CREATE TEMP FUNCTION IS_NI (good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
+  good / (good + needs_improvement + poor) < 0.75 AND
+  poor / (good + needs_improvement + poor) < 0.25
+);
+
+CREATE TEMP FUNCTION IS_POOR (good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
+  poor / (good + needs_improvement + poor) >= 0.25
+);
+
+CREATE TEMP FUNCTION IS_NON_ZERO (good FLOAT64, needs_improvement FLOAT64, poor FLOAT64) RETURNS BOOL AS (
+  good + needs_improvement + poor > 0
+);
+
+WITH
+base AS (
+  SELECT
+    date,
+    origin,
+    device,
+
+    fast_fid,
+    avg_fid,
+    slow_fid,
+
+    fast_lcp,
+    avg_lcp,
+    slow_lcp,
+
+    small_cls,
+    medium_cls,
+    large_cls,
+
+    fast_fcp,
+    avg_fcp,
+    slow_fcp,
+
+    fast_ttfb,
+    avg_ttfb,
+    slow_ttfb
+
+  FROM
+    `chrome-ux-report.materialized.device_summary`
+  WHERE
+    device IN ('desktop', 'phone') AND
+    date IN ('2020-07-01', '2021-07-01')
```

Comment:
@siakaramalegos are you interested in YoY comparisons? If not this could be removed/simplified.
```suggestion
    date = '2021-07-01'
```

Reply:
Yes, I wanted to look at a year-over-year comparison for the vitals where it makes sense (not CLS)

Reply:
The 2020 Almanac was actually a month late so the dataset for comparison should be August (updated suggestion)